File: app/src/main/java/ru/samtakoy/core/screens/SingleFragmentActivity.java
package ru.samtakoy.core.screens;

import android.os.Bundle;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import ru.samtakoy.R;

public abstract class SingleFragmentActivity extends AppCompatActivity {

    protected abstract Fragment createFragment();

    @LayoutRes
    protected int getLayoutResId() {
        return R.layout.activity_single_fragment;
    }

    protected int getFragmentContainerId() {
        return R.id.fragment_cont;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(getLayoutResId());

        FragmentManager fm = getSupportFragmentManager();
        Fragment f = fm.findFragmentById(getFragmentContainerId());
        if(f == null){
            f = createFragment();
            fm.beginTransaction()
                    .add(getFragmentContainerId(), f)
                    .commit();
        }
    }


}

File: app/src/main/java/ru/samtakoy/core/presentation/SingleFragmentActivity.kt
package ru.samtakoy.core.presentation

import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import ru.samtakoy.R

abstract class SingleFragmentActivity : AppCompatActivity() {
    protected abstract fun createFragment(): Fragment

    @get:LayoutRes
    protected val layoutResId: Int
        get() = R.layout.activity_single_fragment

    protected val fragmentContainerId: Int
        get() = R.id.fragment_cont

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(this.layoutResId)

        val fm = getSupportFragmentManager()
        var f = fm.findFragmentById(this.fragmentContainerId)
        if (f == null) {
            f = createFragment()
            fm.beginTransaction()
                .add(this.fragmentContainerId, f)
                .commit()
        }
    }
}
